src/Parser/Monk/Brewmaster/Modules/Spells/KegSmash.js

```diff
@@ -0,0 +1,88 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import BlackoutCombo from './BlackoutCombo';
+import SharedBrews from '../Core/SharedBrews';
+
+const KEG_SMASH_REDUCTION = 4000;
+const BOC_KEG_SMASH_REDUCTION = 2000;
+
+class KegSmash extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    boc: BlackoutCombo,
+    brews: SharedBrews,
+  }
+
+  totalCasts = 0;
+  totalHits = 0;
+  bocHits = 0;
+
+  cdr = 0;
+  bocCDR = 0;
+  wastedCDR = 0;
+  wastedBocCDR = 0;
+
+  _next_target = 0;
+
+  _boc_buff_active = false;
+  
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if(SPELLS.BLACKOUT_COMBO_BUFF.id === spellId) {
+      this._boc_buff_active = true;
+    }
+  }
+
+  on_byPlayer_refreshbuff(event) {
+    const spellId = event.ability.guid;
+    if(SPELLS.BLACKOUT_COMBO_BUFF.id === spellId) {
+      this._boc_buff_active = true;
+    }
+  }
+
+  on_byPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if(SPELLS.BLACKOUT_COMBO_BUFF.id === spellId) {
+      this._boc_buff_active = false;
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if(SPELLS.KEG_SMASH.id === spellId) {
```

Comment:
It's preferred to return early instead of nesting large pieces of code in if-statements. As code grows this quickly becomes hard to understand and maintain.